rust/agents/scraper/migration/src/m20220805_000001_create_table_domain.rs

```diff
@@ -0,0 +1,144 @@
+use std::time;
+use std::time::UNIX_EPOCH;
+
+use sea_orm::prelude::DateTime;
+use sea_orm_migration::prelude::*;
+
+use abacus_core::domain_from_chain;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+/// Chain name, native currency symbol, chain id, is test net
+const DOMAINS: &[(&str, &str, u64, bool)] = &[
+    ("alfajores", "CELO", 44787, true),
+    ("arbitrum", "ETH", 42161, false),
+    ("arbitrumrinkeby", "ETH", 421611, true),
+    ("auroratestnet", "ETH", 1313161555, true),
+    ("avalanche", "AVAX", 43114, false),
+    ("bsc", "BNB", 56, false),
+    ("bsctestnet", "tBNB", 97, true),
+    ("celo", "CELO", 42220, false),
+    ("ethereum", "ETH", 1, false),
+    ("fuji", "AVAX", 43113, true),
+    ("goerli", "ETH", 5, true),
+    ("kovan", "ETH", 42, true),
+    ("mumbai", "MATIC", 80001, true),
+    ("optimism", "ETH", 10, false),
+    ("optimismkovan", "ETH", 69, true),
+    ("polygon", "MATIC", 137, false),
+    ("test1", "ETH", 0, true),
+    ("test2", "ETH", 0, true),
+    ("test3", "ETH", 0, true),
+];
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .create_table(
+                Table::create()
+                    .table(Domain::Table)
+                    .if_not_exists()
+                    .col(
+                        ColumnDef::new(Domain::Id)
+                            .integer()
+                            .not_null()
+                            .auto_increment()
```

Comment:
Actually it can't, fun fact, that was a source of a cryptic error. Using the DomainID caused issues when trying to insert because it had errors deserializing the returned key values. The ORM just does not support having custom IDs like that as far as I can tell.

Reply:
Ha dang, okay good to know